exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/Entity.kt

```diff
@@ -111,16 +111,21 @@ open class Entity<ID:Comparable<ID>>(val id: EntityID<ID>) {
         klass.invalidateEntityInCache(o)
         val currentValue = _readValues?.getOrNull(this)
         if (writeValues.containsKey(this as Column<out Any?>) || currentValue != value) {
+            val entityCache = TransactionManager.current().entityCache
             if (referee != null) {
-                val entityCache = TransactionManager.current().entityCache
                 if (value is EntityID<*> && value.table == referee!!.table) value.value // flush
 
                 listOfNotNull<Any>(value, currentValue).forEach {
-                    entityCache.referrers[it]?.remove(this)
+                    entityCache.referrers[this]?.remove(it)
                 }
                 entityCache.removeTablesReferrers(listOf(referee!!.table))
             }
             writeValues[this as Column<Any?>] = value
+
+            // TODO: Can this be simplified?
+            if (o.id._value != null && entityCache.data[table]?.contains(id._value) == true) {
```

Comment:
This forces updates to behave like [this](https://github.com/JetBrains/Exposed/pull/1133/files#diff-b16f6a7584869bec39e5e0822151a2131c95b4eca29d8e85881e4ff6489313e6L53) was before.  I attempted to use `isNewEntity` (backed by `inserts`), but depending on if the `id` is null or not the entity may or may not be present in `inserts`.  However the entity is only added to `data` after it has been inserted, and is therefore available for updating.  I don't like the syntax, but it works.